### ISSUE TITLE
加载训练的word2vec嵌入向量问题

### ISSUE THREAD

**Issue by PuHaoran** [Magic-Bubble/Zhihu#2]
你好前辈，请问
char_embed_file = '../ieee_zhihu_cup/char_embedding.txt'
word_embed_file = '../ieee_zhihu_cup/word_embedding.txt'
char_alphabet, char_embed_mat, unknown_char_idx, dummy_char_idx = parse_embed(char_embed_file)
word_alphabet, word_embed_mat, unknown_word_idx, dummy_word_idx = parse_embed(word_embed_file)
中char_embed_file这个文件是如何得到的，代码在程序中有吗。这个文件里的内容是如何生成的呢？

**Comment by Magic-Bubble** [comment]
这个是知乎比赛提供的原始数据，是他们预训练好的～

**Comment by PuHaoran** [comment]
tks，感谢 感谢。再问个问题。fastText模型在Embedding层之后有个TimeDistributed，这快看的头大，你的三个模型里都有这一步，但我实在没看懂它的作用。可以详细讲解下吗 不胜感激啊
        self.tdfc1 = nn.Linear(D, 512)
        self.td1 = TimeDistributed(self.tdfc1)
        self.tdbn1 = nn.BatchNorm2d(1)
        
        self.tdfc2 = nn.Linear(D, 512)
        self.td2 = TimeDistributed(self.tdfc2)
        self.tdbn2 = nn.BatchNorm2d(1)
        
        self.fc1 = nn.Linear(1024, 512)
        self.bn1 = nn.BatchNorm1d(512)
        self.fc2 = nn.Linear(512, C)
....
    def forward(self, x, y):
        if self.opt['use_char_word']:
            x = self.embed_char(x.long())
            y = self.embed_word(y.long())
        elif self.opt['use_word_char']:
            x = self.embed_word(x.long())
            y = self.embed_char(y.long())
        else:
            x = self.embed(x.long())
            y = self.embed(y.long())

        if self.opt['static']:
            x = x.detach()
        x = F.relu(self.tdbn1(self.td1(x).unsqueeze(1))).squeeze(1)
            
        if self.opt['static']:
            y = y.detach()
        y = F.relu(self.tdbn2(self.td2(y).unsqueeze(1))).squeeze(1)
            
        x = x.mean(1).squeeze(1)
        y = y.mean(1).squeeze(1)
        
        x = torch.cat((x, y), 1)

        x = F.relu(self.bn1(self.fc1(x)))
        logit = self.fc2(x)
        return logit

**Comment by Magic-Bubble** [comment]
可以理解为对原始的embedding做一个transform，是一层timestep的fc，会提升模型效果
